tests/buildAndTest.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+
+# automated build and test script for the datacompressor.
+# the functions containing the actual tests can be found in separate scripts in the same directory
+# the emulator function is used by the testfunctions and also contained in another file.
+
+set -euo pipefail
+
+workdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+source "$workdir/emulator.sh"
+source "$workdir/test_dega.sh"
+source "$workdir/test_lzmh.sh"
+source "$workdir/test_copy.sh"
+
+dataCompressorPath="../data-compressor"
+
+is_numeric_regex='^[0-9]+$'
+
+linux_dccli_sep="\#"
+windows_dccli_sep="\#"
+
+linux_dccli_build_dir="$dataCompressorPath/DataCompressor/build/gcc/"
+linux_dccli_command="$dataCompressorPath/DataCompressor/build/gcc/DCCLI"
+
+testdata_input="$dataCompressorPath/DataCompressor/DCCLI/testdata/input.txt"
+testdata_output="$dataCompressorPath/DataCompressor/DCCLI/testdata/output.txt"
+
+windows_compiler="/c/Program\ Files\ \(x86\)/MSBuild/12.0/Bin/MSBuild.exe"
+windows_project_file="$dataCompressorPath/DataCompressor/build/MSVC/DataCompressor.sln"
+win32_dccli_command="$dataCompressorPath/DataCompressor/build/MSVC/Release/DCCLI.exe"
+win_x64_dccli_command="$dataCompressorPath/DataCompressor/build/MSVC/x64/Release/DCCLI.exe"
+
+script_name="$(basename "$0")"
+
+success="success"
+
+err_msg() {
+        echo "Error: $1"
+        echo "Linux: $script_name <i386|x86_64|armhf> <IO_SIZE_BITS>"
+        echo "Raspberry: $script_name <arm> <IO_SIZE_BITS>"
+        echo "Windows: $script_name <win32|x64> <IO_SIZE_BITS>"
+}
+
+tests() {
+        # arg 1: IO_SIZE_BITS for copy blocksize
+        if ! [ $# -eq 1 ] || ! [[ $1 =~ $is_numeric_regex ]]; then
+                echo "tests: exactly one numeric argument expected"
+                exit 1
+        fi
+
+        local result_dega result_lzmh result_copy
+        echo "=========================================================="
+        echo "testing DEGA"
+        echo "=========================================================="
+        test_dega result_dega
+        echo "=========================================================="
+        echo "testing LZMH"
+        echo "=========================================================="
+        test_lzmh result_lzmh
+        echo "=========================================================="
+#        echo "testing copy"
+#        echo "=========================================================="
+#        test_copy "$1" result_copy
+#        echo "=========================================================="
+#        echo "test results"
+#        echo "=========================================================="
+        echo "dega: $result_dega"
+        echo "lzmh: $result_lzmh"
+#        echo "copy: $result_copy"
```

Comment:
To be removed?

Reply:
No. Currently, due to unresolved issues in the Data Compressor, the tests for the copy module fail for some blocksizes. This part was commented out because it made troubleshooting during the development of the test environment easier. Removed the comments now.